lemur/plugins/lemur_gcp/plugin.py

```diff
@@ -0,0 +1,107 @@
+from flask import current_app
+from google.cloud.compute_v1.services import ssl_certificates
+from google.oauth2 import service_account
+import google.auth
+import hvac
+import os
+
+from lemur.plugins.bases import DestinationPlugin
+from lemur.plugins import lemur_gcp as gcp
+
+
+class GCPDestinationPlugin(DestinationPlugin):
+    title = "GCP"
+    slug = "gcp-destination"
+    version = gcp.VERSION
+    description = "Allow the uploading of certificates to GCP"
+    author = "Mitch Cail"
+    author_url = "https://github.com/Datadog/lemur"
+
+    options = [
+        {
+            "name": "accountId",
+            "type": "str",
+            "required": True,
+            "helpMessage": "GCP Project ID",
+        },
+        {
+            "name": "authenticationMethod",
+            "type": "select",
+            "required": True,
+            "available": ["vault", "serviceAccountToken"],
+            "helpMessage": "Authentication method to use",
+            "default": False,
+        },
+        {
+            "name": "vaultMountPoint",
+            "type": "str",
+            "required": False,
+            "helpMessage": "Path to vault secret",
+        },
+        {
+            "name": "serviceAccountTokenPath",
+            "type": "str",
+            "required": False,
+            "helpMessage": "Path to vault secret",
+        }
+    ]
+
+    def upload(self, name, body, private_key, cert_chain, options, **kwargs):
+        """
+        Args:
+        certificate_file: path to the file with the certificate you want to create in your project.
+        private_key_file: path to the private key you used to sign the certificate with.
+        certificate_name: name for the certificate once it's created in your project.
+
+        """
+
+        try:
+            ssl_certificate_body = {
+                "name": name,
+                "certificate": body,
+                "description": "",
+                "private_key": private_key,
+            }
+            credentials = self._get_gcp_credentials(options)
+            return self._insert_gcp_certificate(
+                self.get_option("AccountId", options),
```

Comment:
This `AccountId` option doesn't seem to exist (the lookup is case-sensitive) so I don't think this will work properly. (We can and should add a test case for this.)